webapp/src/main/java/com/box/l10n/mojito/entity/PollableTask.java

```diff
@@ -38,6 +40,7 @@
                 @Index(name = "I__POLLABLE_TASK__FINISHED_DATE", columnList = "finished_date")
         }
 )
+@BatchSize(size = 1000)
```

Comment:
- 1 is the default batch size and N+1 is a solution to the cartesian product explosion problem.

&nbsp;&nbsp;&nbsp;&nbsp;The only reason to default to 1 instead of say, 10 or 10000, is potential memory limitations and those would only be seen at runtime and highly dependent on each application built on top of Hibernate (hence why the only safe default is 1). 
&nbsp;&nbsp;&nbsp;&nbsp;I.e.: In one application, you could, in principle, have each row/entity you fetch from the DB be a 100 MB JSON document (not saying that's a good design..) and the max heap size could be configured in that case to 256 MB. Any batch size about _two_ would result in out of memory exceptions. 
Hence why you can only pick batch sizes based on your application and specific entities that you know the size and access patterns for.

- Happy to detail the batch size as well as entity selection in more detail, most of it was repeated iteration on large data sets based on actual branchStatistic API queries and analysing all of the issued queries. I've reverted some entities back to eager loading so the batch size doesn't (yet) have an impact in those cases - though I believe there might still be some value in having batching configured for them - I wanted to discuss that with you.